pkg/apis/application/v1alpha1/app_project_types.go

```diff
@@ -252,12 +252,13 @@ func (proj *AppProject) ValidateProject() error {
 	}
 
 	if proj.Spec.SyncWindows.HasWindows() {
-		existingWindows := make(map[string]bool)
+		existingWindows := make(map[uint64]bool)
 		for _, window := range proj.Spec.SyncWindows {
 			if window == nil {
 				continue
 			}
-			if _, ok := existingWindows[window.Kind+window.Schedule+window.Duration]; ok {
+			windowHash := window.Hash()
+			if _, ok := existingWindows[windowHash]; ok {
 				return status.Errorf(codes.AlreadyExists, "window '%s':'%s':'%s' already exists, update or edit", window.Kind, window.Schedule, window.Duration)
```

Comment:
Let's add the hash to the message as well

Reply:
Okay, sure 😄